src/telegramServer.ts

```diff
@@ -217,29 +217,23 @@ export class TelegramServer extends EventEmitter {
     }
   }
 
-  /**
-   * @FIXME
-   * (node:103570) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11
-   * EditedMessageText listeners added to [TelegramServer]. Use emitter.setMaxListeners() to
-   * increase limit (Use `node --trace-warnings ...` to show where the warning was created)
-   */
   async waitBotEdits() {
     return new Promise<void>((resolve) => {
-      this.on('EditedMessageText', () => resolve());
+      this.once('EditedMessageText', () => resolve());
     });
   }
 
   async waitBotMessage() {
     return new Promise<void>((resolve) => {
-      this.on('AddedBotMessage', () => resolve());
+      this.once('AddedBotMessage', () => resolve());
     });
   }
 
   async waitUserMessage() {
     return new Promise<void>((resolve) => {
-      this.on('AddedUserMessage', () => resolve());
-      this.on('AddedUserCommand', () => resolve());
-      this.on('AddedUserCallbackQuery', () => resolve());
+      this.once('AddedUserMessage', () => resolve());
```

Comment:
Everything was right here in previous commit version. We wait for any signal and the turn off all emitters and resolve promise.

Now, one emitter will turn itself off but other two will stay and that's not what we expect :)

Reply:
Ahh, that's why you "can't think of a better way too"! =)) Indeed!

Reply:
Yup. Emitter is a great power which comes with great possibility of shooting your legs off :)